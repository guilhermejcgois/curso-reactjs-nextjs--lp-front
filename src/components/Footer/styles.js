import style, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = style.p`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
