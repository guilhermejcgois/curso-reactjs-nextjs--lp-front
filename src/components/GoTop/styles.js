import styled from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => `
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-contener: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
