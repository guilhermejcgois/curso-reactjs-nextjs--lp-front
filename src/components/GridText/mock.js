export default {
  background: false,
  title: 'My grid',
  description:
    'Etiam a sapien vitae nisi accumsan aliquet. Quisque pellentesque facilisis lacus, eu iaculis ligula cursus eget.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
  ],
};
