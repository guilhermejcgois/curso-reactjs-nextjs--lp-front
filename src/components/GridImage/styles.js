import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
