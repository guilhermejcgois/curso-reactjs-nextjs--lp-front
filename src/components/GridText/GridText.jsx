import React from 'react';
import P from 'prop-types';
import { Heading } from '../Heading';
import { SectionBackground } from '../SectionBackground';
import { TextComponent } from '../TextComponent';
import * as styled from './styles';

export const GridText = ({ title, description, grid, background = false }) => (
  <SectionBackground background={background}>
    <styled.Container>
      <Heading size="huge" colorDark={!background} uppercase as="h2">
        {title}
      </Heading>
      <TextComponent>{description}</TextComponent>
      <styled.Grid>
        {grid.map((el, i) => (
          <styled.GridElement key={i}>
            <Heading size="medium" colorDark={!background} as="h3">
              {el.title}
            </Heading>
            <TextComponent>{el.description}</TextComponent>
          </styled.GridElement>
        ))}
      </styled.Grid>
    </styled.Container>
  </SectionBackground>
);

GridText.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }),
  ).isRequired,
  background: P.bool,
};
