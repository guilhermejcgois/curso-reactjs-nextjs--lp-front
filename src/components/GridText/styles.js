import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position absolute;
      font-size: ${theme.font.sizes.xhuge};
      top -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
