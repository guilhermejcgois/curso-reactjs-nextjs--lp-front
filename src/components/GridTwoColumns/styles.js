import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacing.xlarge};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;
