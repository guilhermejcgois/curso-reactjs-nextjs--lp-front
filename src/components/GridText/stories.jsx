import React from 'react';
import { GridText } from './GridText';
import mock from './mock';

export default {
  title: 'GridText',
  component: GridText,
  args: mock,
  argTypes: {
    title: { type: 'string' },
    html: { type: 'string' },
    background: { type: 'boolean' },
  },
};

export const Template = (args) => <GridText {...args} />;
