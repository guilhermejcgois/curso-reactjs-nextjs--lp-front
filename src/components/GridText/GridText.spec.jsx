import React from 'react';
import { renderTheme } from '../../styles/render-theme';
import { GridText } from './GridText';
import mock from './mock';

describe('<GridText />', () => {
  it('should render grid section with background', () => {
    const { container } = renderTheme(<GridText {...mock} background={true} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid section without background', () => {
    const { container } = renderTheme(<GridText {...mock} background={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
