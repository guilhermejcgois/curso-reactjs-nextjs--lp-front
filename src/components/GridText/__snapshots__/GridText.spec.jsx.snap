// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> should render grid section with background 1`] = `
.c4 {
  color: #ffffff;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #ffffff;
  font-size: 3.2rem;
  text-transform: none;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  background: #0a1128;
  color: #ffffff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

@media (max-width:768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <div
          class="c5 c6"
        >
          Etiam a sapien vitae nisi accumsan aliquet. Quisque pellentesque facilisis lacus, eu iaculis ligula cursus eget.
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridText /> should render grid section without background 1`] = `
.c4 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0a1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

@media (max-width:768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          My grid
        </h2>
        <div
          class="c5 c6"
        >
          Etiam a sapien vitae nisi accumsan aliquet. Quisque pellentesque facilisis lacus, eu iaculis ligula cursus eget.
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 1
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 2
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c3 c9"
            >
              Teste 3
            </h3>
            <div
              class="c5 c6"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
